mne/forward/_field_interpolation.py

```diff
@@ -132,35 +140,62 @@ def _map_meg_channels(info_from, info_to, mode='fast', origin=(0., 0., 0.04)):
 
     Returns
     -------
-    mapping : array
-        A mapping matrix of shape len(pick_to) x len(pick_from).
+    mapping : array, shape (n_to, n_from)
+        A mapping matrix.
     """
     # no need to apply trans because both from and to coils are in device
     # coordinates
-    templates = _read_coil_defs(verbose=False)
-    coils_from = _create_meg_coils(info_from['chs'], 'normal',
-                                   info_from['dev_head_t'], templates)
-    coils_to = _create_meg_coils(info_to['chs'], 'normal',
-                                 info_to['dev_head_t'], templates)
-    miss = 1e-4  # Smoothing criterion for MEG
+    info_kinds = set(ch['kind'] for ch in info_to['chs'])
+    info_kinds |= set(ch['kind'] for ch in info_from['chs'])
+    if FIFF.FIFFV_REF_MEG_CH in info_kinds:  # refs same as MEG
+        info_kinds |= set([FIFF.FIFFV_MEG_CH])
+        info_kinds -= set([FIFF.FIFFV_REF_MEG_CH])
+    info_kinds = sorted(info_kinds)
+    # This should be guaranteed by the callers
+    assert (len(info_kinds) == 1 and info_kinds[0] in (
+            FIFF.FIFFV_MEG_CH, FIFF.FIFFV_EEG_CH))
+    kind = 'eeg' if info_kinds[0] == FIFF.FIFFV_EEG_CH else 'meg'
+
+    #
+    # Step 1. Prepare the coil definitions
+    #
+    if kind == 'meg':
+        templates = _read_coil_defs(verbose=False)
+        coils_from = _create_meg_coils(info_from['chs'], 'normal',
+                                       info_from['dev_head_t'], templates)
+        coils_to = _create_meg_coils(info_to['chs'], 'normal',
+                                     info_to['dev_head_t'], templates)
+        miss = 1e-4
+        pinv_method = 'tsvd'
+    else:
+        coils_from = _create_eeg_els(info_from['chs'])
+        coils_to = _create_eeg_els(info_to['chs'])
+        miss = 1e-2
+        pinv_method = 'tikhonov'
```

Comment:
This inversion method was chosen because it worked better (higher correlation, lower difference values) in testing compared to the `miss=1e-3, pinv_method='tsvd'` that was the default in the C code

Reply:
on how many datasets did you try? can you try with one of the EEG only dataset we have?

Reply:
Okay wrote a [little script](https://gist.github.com/larsoner/da24fbc58c39ffd9c09e278aa0a26c32) to test kiloword, LIMO, EEGBCI, and sample. Results for these four (plus average in the last row) were:
```
|  sph   |                        TSVD                         |                           Tikhonov                           |
| 1e-05  | 5e-01  | 1e-01  | 1e-02  | 1e-03  | 1e-04  | 1e-05  | 1e+00  | 5e-01  | 1e-01  | 1e-02  | 1e-03  | 1e-04  | 1e-05  |
|--------|--------|--------|--------|--------|--------|--------|--------|--------|--------|--------|--------|--------|--------|
| 0.9086 | 0.7167 | 0.9067 | 0.9082 | 0.8694 | 0.8287 | 0.8287 | 0.9216 | 0.9255 | 0.9215 | 0.8964 | 0.8414 | 0.8310 | 0.8289 |
| 0.7701 | 0.6763 | 0.7699 | 0.7646 | 0.7563 | 0.7046 | 0.6394 | 0.7799 | 0.7837 | 0.7870 | 0.7596 | 0.7104 | 0.6567 | 0.6169 |
| 0.9238 | 0.6771 | 0.8275 | 0.8886 | 0.8806 | 0.8627 | 0.7838 | 0.8861 | 0.8993 | 0.9212 | 0.9071 | 0.8771 | 0.8389 | 0.7572 |
| 0.5215 | 0.4252 | 0.5873 | 0.5310 | 0.5535 | 0.6519 | 0.6906 | 0.5342 | 0.5304 | 0.5338 | 0.4847 | 0.5819 | 0.7047 | 0.7198 |
|--------|--------|--------|--------|--------|--------|--------|--------|--------|--------|--------|--------|--------|--------|
| 0.7810 | 0.6238 | 0.7729 | 0.7731 | 0.7649 | 0.7620 | 0.7356 | 0.7804 | 0.7847 | 0.7909 | 0.7619 | 0.7527 | 0.7578 | 0.7307 |
```
So I guess Tikhonov with 1e-1 reg is the winner?